### ISSUE TITLE
MD005 Inconsistent indentation for list items at the same level for ordered lists

### ISSUE THREAD

**Issue by benma** [markdownlint/markdownlint#139]
Hi

```
1. List Item
  1. Sub List Item
  1. Sub List Item
```

```
$ mdl test.md
test.md:2: MD005 Inconsistent indentation for list items at the same level
test.md:3: MD005 Inconsistent indentation for list items at the same level
```

For unordered lists, there is no linter error. 

If the sub items are indented with 4 spaces, there is no issue, but I'd like the indentation levels to be consistent in the doc.

**Comment by mivok** [comment]
The parser markdownlint uses, kramdown, currently requires that ordered lists be indented by at least 3 spaces in order for them to be recognized as a different level (the digit, the following dot, and then a space after it). I'll need to investigate if this is a bug in kramdown, or simply a difference in behavior between markdown implementations. As it is, in at least one parser, a numbered list like that is actually considered to be the same level.

Right now I'm not sure what a proper fix for this will be, but at the very least there should be a rule warning that indenting by just two spaces won't work in all parsers.

Interestingly, the github parser seems to work differently, with one space being necessary for the first level indent, and two additional spaces for any sub lists afterwards:
1. No indent
   1. One space indent
   2. Two space indent
   3. Three space indent
   4. Four space indent
2. No indent
   1. 2 spaces
      1. 4 spaces
3. No indent
   1. 3 spaces
      1. 6 spaces
4. No indent
   1. One space
      1. 3 spaces

**Comment by DavidAnson** [comment]
This may be relevant:
https://github.com/DavidAnson/markdownlint/issues/20

**Comment by FranklinYu** [comment]
I think 4-space indentation is natural choice, because ordered list marker takes 3 spaces. Compare current spec with

```markdown
1.  First item

    1.  first sub item
    2.  second sub item

    description for block code

        char[] block = "quote";

    summary for block quote

2.  second item

    *   first unordered list
    *   second unordered list

12. This can be nice up to 99
```

rendered as

> 1.  First item
>
>     1.  first sub item
>     2.  second sub item
>
>     description for block code
>
>         char[] block = "quote";
>
>     summary for block quote
>
> 2.  second item
>
>     *   first unordered list
>     *   second unordered list
>
> 12. This can be nice up to 99

**Comment by jaymzh** [comment]
I think the right solution here, frankly is to set the default for MD007 to be 3 instead of 2. Because it's impossible (given how kramdown works) for us to recognize 2-space indentations as nest. I'll do that.
